cmd/termcord/main.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"log"
+
+	"github.com/haguro/termcord/pkg/termcorder"
+)
+
+func main() {
+	tc, err := termcorder.TermcordingFromFlags()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := tc.Start(); err != nil {
```

Comment:
Having to check two errors is a bit of a red flag here, isn't it? If we _have_ a valid termcorder (and the fact that the constructor returns error would suggest we do), then how come we can't start it successfully?

It's always a nice idea (where possible) to have your constructor produce not only a _valid_ object, but also a _usable_ one. This doesn't always make sense, but I think we could do it here, couldn't we?

Reply:
Agreed! However the argument for `Start` returning an error is that a user (of the library) might not use the constructor to create a `termcorder` value and as such they might not have configured it properly.

To make sure I understand you correctly though, did you mean that we don't need to check for the error in the particular use case in `main`, or did you mean that `Start` shouldn't be returning an error?

Reply:
Noting this here for later maybe: `Start` does a couple of important things, namely it _starts_ the command inside a psuedo-terminal (by calling `pseudoTermFromCmd` - which in turn has a couple of chances of returning error) and also sets up the writers `tc.in` and `tc.out (by calling `setupWriters` - which also has the potential to return an error, for example if it cannot open a recording file for writing). The only way I see `Start` not having to return any errors is if we were to move all that work to the constructor. That doesn't look like a good idea to me especially since the command is actually started inside the pty in `pseudoTermFromCmd`.